packages/engine/src/worker/runner/python/mod.rs

```diff
@@ -141,8 +141,9 @@ async fn _run(
             Some(nng_send_result) = nng_sender.get_send_result() => {
                 nng_send_result?;
             }
-            Some((sim_id, inbound)) = inbound_receiver.recv() => {
-                let (task_payload_json, task_wrapper) = match &inbound {
+            Some((_sim_id, inbound)) = inbound_receiver.recv() => {
+                // Need to get payload before sending nng message.
+                let (_task_payload_json, task_wrapper) = match &inbound {
                     InboundToRunnerMsgPayload::TaskMsg(msg) => {
                         // TODO: Error message duplication with JS runner
```

Comment:
This says JS runner, but this is in `runner/python`, copy pasted from the JS impl perhaps? Or is it just necessary because of JS?
Is the duplication referring to the two payloads here?

Reply:
Sorry I don't really understand. Are you saying it's code duplication that could be abstracted out and both runners can rely on it?

Reply:
Correct, both runners have a similar piece of logic